integration/postgres.go

```diff
@@ -60,17 +61,21 @@ func (db testPostgresDB) Reset() error {
 		return err
 	}
 	defer dbSQL.Close()
-	resetQuery := "DROP SCHEMA public CASCADE;" +
-		"CREATE SCHEMA public;" +
-		"GRANT ALL ON SCHEMA public TO postgres;" +
-		"GRANT ALL ON SCHEMA public TO public;"
-	// drop all the tables in the test database
-	if _, err := dbSQL.Exec(resetQuery); err != nil {
-		return err
+	if db.migrateDownFunction != nil {
+		db.migrateDownFunction(dbSQL)
+	} else {
+		resetQuery := "DROP SCHEMA public CASCADE;" +
```

Comment:
nitpick: you could instead have this be the default `db.migrateDownFunction` when calling `NewTestPostgresDB` and get rid of this `else` statement. Not a big deal, just pointing it out